File: deduba_os.cc
#include "deduba_os.hh"

#define EXPORT(func) exports.Set(Napi::String::New(env, #func), Napi::Function::New(env, do_##func));

Napi::Object Init(Napi::Env env, Napi::Object exports)
{
  EXPORT(BZ2_bzWrite);
  EXPORT(BZ2_bzWriteClose);
  EXPORT(BZ2_bzWriteOpen);
  EXPORT(getgrgid);
  EXPORT(getpwuid);
  return exports;
}

NODE_API_MODULE(deduba_os, Init)

File: deduba_os.js
const deduba_os = require("./prebuilds/linux-x64/node.napi");

module.exports = { getpwuid: deduba_os.getpwuid, getgrgid: deduba_os.getgrgid };

function testuser(u) {
  usr = deduba_os.getpwuid(u);
  grp = deduba_os.getgrgid(usr.gid);
  console.log({ usr, grp });
}
testuser(0);
testuser(1000);
testuser(152);
testuser();
